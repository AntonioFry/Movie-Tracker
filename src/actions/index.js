export const showMovies = (movies) => ({
  type: 'SHOW_MOVIES',
  movies
});