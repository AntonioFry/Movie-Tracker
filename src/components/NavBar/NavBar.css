.header {
    display: grid;
    grid-template-columns: 1fr 1fr .5fr .5fr;
    grid-template-rows: 1fr;
    grid-column-gap: 0px;
    grid-row-gap: 0px;
}
.form {
  grid-area: 1 / 1 / 2 / 2;
}
.nav-fav {
  grid-area: 1 / 3 / 2 / 4; 
  margin: auto 0 ;
}

.nav-sign-up {
  grid-area: 1 / 4 / 2 / 5;
  margin: auto 0;
}

.nav-btn {
  background-color: black;
  border-color: red;
  color: red;
  font-size: 1.5rem;
  width: 90%;
}

.nav-btn:hover {
  background-color: red;
  border-color: black;
  color: black;
}
.app-name {
  color:red;
  display: flex;
  grid-area: 1 / 2 / 2 / 3;
  justify-content: center;
  font-size: 4rem;
}

.bang {
  color: white;
}
