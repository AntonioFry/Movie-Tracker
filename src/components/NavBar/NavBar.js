import React from 'react';
import {Route, NavLink} from 'react-router-dom'
import MovieContainter from '../MovieContainer/MovieContainer'
import MovieSpecs from '../MovieSpecs/MovieSpecs'
import AccountMenu from '../AccountMenu/AccountMenu'

const NavBar = ({movies}) => {
  return (
    <div className = 'header'>
      <NavLink to='/' className='nav'>Movies</NavLink>
      <NavLink to='/favorites' className='nav'>Favorites</NavLink>
      <AccountMenu />
      <Route exact path='/' render={() => <MovieContainter movies={movies}/>} />
      <Route exact path='/:id' render={({match}) => {
        const {id} = match.params;
        const description = movies.find(movie => {
          return movie.id === parseInt(id)
        });
        return description && <MovieSpecs {...description} />
      }}/>
      <Route exact path='/Favorites' component={MovieContainter}/>
    </div>
  )
}

export default NavBar;