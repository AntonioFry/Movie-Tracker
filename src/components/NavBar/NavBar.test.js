import React from 'react';
import { shallow } from 'enzyme';
import { NavBar, mapStateToProps, mapDispatchToProps } from './NavBar';
import { logOut, setFavorites } from '../../actions';

describe('NavBar', () => {
  let wrapper;

  const mockMovies = [
    {
      vote_count:411,
      id:420818,
      video:false,
      vote_average:7,
      title:"The Lion King",
      popularity:585.503,
      poster_path:"/dzBtMocZuJbjLOXvrl4zGYigDzh.jpg",
      original_language:"en",
      original_title:"The Lion King",
      backdrop_path:"/1TUg5pO1VZ4B0Q1amk3OlXvlpXV.jpg",
      adult:false,
      overview:"Simba idolises his father, King Mufasa, and takes to heart his own royal destiny. But not everyone in the kingdom celebrates the new cub's arrival. Scar, Mufasa's brother—and former heir to the throne—has plans of his own. The battle for Pride Rock is ravaged with betrayal, tragedy and drama, ultimately resulting in Simba's exile. With help from a curious pair of newfound friends, Simba will have to figure out how to grow up and take back what is rightfully his.",
      release_date:"2019-07-12"}
  ]

  const mockUser = {
    id: 1,
    name: 'Andre 3000',
    password: 'imBetterThanBiGBOI324',
    email: 'blah@blah.com'
  }
  beforeEach(() => {
    wrapper = shallow(
      <NavBar
      movies={mockMovies}
      user={mockUser}
      />
    )
  });
  it('should match snapshot',() => {
    expect(wrapper).toMatchSnapshot() 
  });
  
  
  describe('mapStateToProps', () => {
    it('should return an array with the users favorite moveis', () => {
      const mockState = {
        userFavorites: [{mockMovies}],
        filter: 'SET_FAVORITES'
      }
      const expected ={
        userFavorites: [{mockMovies}]
      }
      const mappedProps = mapStateToProps(mockState)
  
      expect(mappedProps).toEqual(expected)
    });
  }); 
  describe('mapDispatchToProps', () => {
    it('should call logout dispatch when logout is clicked', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = logOut();
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.logout();

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('should call setFavorites when logout is clicked', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = setFavorites();
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setFavorites();

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    });
  });
}); 