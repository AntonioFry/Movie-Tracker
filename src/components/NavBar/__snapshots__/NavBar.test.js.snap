// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match snapshot 1`] = `
<div>
  <div
    className="header"
  >
    <h1
      className="app-name"
    >
       
      <span
        className="bang"
      >
        !
      </span>
      Netflix
    </h1>
    <NavLink
      className="nav-fav"
      to="/favorites"
    >
      <button
        className="nav-btn"
      >
        Favorites
      </button>
    </NavLink>
    <NavLink
      className="nav-log-in"
      to="/Login"
    >
      <button
        className="nav-log-in-btn"
      >
        Login
      </button>
    </NavLink>
    <NavLink
      className="nav-sign-up"
      to="/signup"
    >
      <button
        className="nav-btn"
      >
        Sign Up
      </button>
    </NavLink>
  </div>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/Login"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/LoggedIn"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/:id"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/Favorites"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/signup"
    render={[Function]}
  />
</div>
`;
